scipy/stats/_sensitivity_analysis.py

```diff
@@ -0,0 +1,717 @@
+        except KeyError as exc:
+            message = (
+                f"{method!r} is not a valid 'method'. It must be one of"
+                f" {set(indices_methods)!r} or a callable."
+            )
+            raise ValueError(message) from exc
+    else:
+        kind = inspect.Parameter.POSITIONAL_OR_KEYWORD
+        parameters = (
+            inspect.Parameter('f_A', kind=kind),
+            inspect.Parameter('f_B', kind=kind),
+            inspect.Parameter('f_AB', kind=kind)
+        )
+        sig = inspect.Signature(parameters=parameters)
+        indices_method_ = method
+
+        if inspect.signature(indices_method_) != sig:
+            message = (
+                "If 'method' is a callable, it must have the following"
+                f" signature: {inspect.signature(saltelli_2010)}"
+            )
+            raise ValueError(message)
+
+    def indices_method(f_A, f_B, f_AB):
+        """Wrap indices method to ensure proper output dimension.
+
+        1D when single output, 2D otherwise.
+        """
+        return np.squeeze(indices_method_(f_A=f_A, f_B=f_B, f_AB=f_AB))
+
+    if callable(func):
+        if dists is None:
+            raise ValueError(
+                "'dists' must be defined when 'func' is a callable."
+            )
+
+        def wrapped_func(x):
+            return np.atleast_2d(func(x))
+
+        A, B = sample_A_B(n=n, dists=dists, random_state=random_state)
+        AB = sample_AB(A=A, B=B)
+
+        f_A = wrapped_func(A)
+
+        if f_A.shape[1] != n:
+            raise ValueError(
+                "'func' output should have a shape ``(s, -1)`` with ``s`` "
+                "the number of output."
+            )
+
+        def funcAB(AB):
+            d, d, n = AB.shape
+            AB = np.moveaxis(AB, 0, -1).reshape(d, n*d)
+            f_AB = wrapped_func(AB)
+            return np.moveaxis(f_AB.reshape((-1, n, d)), -1, 0)
+
+        f_B = wrapped_func(B)
+        f_AB = funcAB(AB)
+    else:
+        message = (
+            "When 'func' is a dictionary, it must contain the following "
+            "keys: 'f_A', 'f_B' and 'f_AB'."
+            "'f_A' and 'f_B' should have a shape ``(s, n)`` and 'f_AB' "
+            "should have a shape ``(d, s, n)``."
+        )
+        try:
+            f_A, f_B, f_AB = np.atleast_2d(
+                func['f_A'], func['f_B'], func['f_AB']
+            )
+        except KeyError as exc:
+            raise ValueError(message) from exc
+
+        if f_A.shape[1] != n or f_A.shape != f_B.shape or \
+                f_AB.shape == f_A.shape or f_AB.shape[-1] % n != 0:
+            raise ValueError(message)
+
+    # Normalization by mean
+    # Sobol', I. and Levitan, Y. L. (1999). On the use of variance reducing
+    # multipliers in monte carlo computations of a global sensitivity index.
+    # Computer Physics Communications, 117(1) :52-61.
+    mean = np.mean([f_A, f_B], axis=(0, -1)).reshape(-1, 1)
```

Comment:
I read the code again, and I think that subtracting the mean is useless here. This is because the estimator which is implemented seems robust to me. Indeed, the particular estimator implemented in lines 147-152 is not sensitive to an offset because you use the `var` function and differences of function values. You do not use the naive the difference of squared that I thought. This is what @andreasaltelli meant, I guess when he wrote "Should not happen". Now that you have a unit test on this topic, I suggest this : remove the code and see if the test pass. If it does, please remove those lines. If it fails, keep the code.

Reply:
Ok thank you, I will test that 👍 Although if this is not harmful, I could as well leave it here in case a user provide an estimator which would not be robust from that perspective.

Otherwise does the rest looks good to you? Especially with regards to the questions Matt had?

Reply:
I can confirm that the test would indeed pass. But I tried to use another formulation (the one we have in the tests) and it would fail. So I think it's good to keep this in.